Source/RpmAvatarCreator/Public/UI/RpmBadgeIconUI.h

```diff
@@ -0,0 +1,17 @@
+// Copyright © 2023++ Ready Player Me
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "Blueprint/UserWidget.h"
+#include "RpmBadgeIconUI.generated.h"
+
+UCLASS(Abstract)
+class RPMAVATARCREATOR_API URpmBadgeIconUI : public UUserWidget
+{
+	GENERATED_BODY()
+
+public:
+	UFUNCTION(BlueprintImplementableEvent, Category = "Ready Player Me", meta = (DisplayName = "Set Icon Texture"))
+	void SetIconTexture(UTexture2D* Image);
```

Comment:
There are multiple classes that have this SetIconTexture function. 
Is the logic for each of the implementations the same? 
I don't suppose we could use interface or inheritance to remove some code duplication?